Compile binaries for access plugin tests

justinas opened the pull request:
Modifies the "Unit tests (Integrations)" workflow to use teleport / tctl compiled from the current commit, rather than downloading the specified version.

Some of the tests require the Enterprise binaries and as such are skipped in the OSS test runs. This is covered by running access plugin tests in `teleport.e` as well, see Enterprise counterpart https://github.com/gravitational/teleport.e/pull/1661

Fixes https://github.com/gravitational/teleport/issues/27528 .

Commit:
Compile binaries for access plugin tests

Commit:
Fix PD test asserts for newer Teleport versions

Commit:
Unpin role from V6 in access plugin tests

justinas commented on integrations/access/pagerduty/pagerduty_test.go:
```diff
@@ -596,7 +596,12 @@ func (s *PagerdutySuite) assertNewEvent(watcher types.Watcher, opType types.OpTy
 			assert.Equal(t, resourceKind, ev.Resource.GetKind())
 			assert.Equal(t, resourceName, ev.Resource.GetName())
 		} else {
-			assert.Nil(t, ev.Resource)
```
Teleport >= 13 sends a `*WatchStatusV1` in `OpInit`. We previously ran tests with 12.x, so we did not hit this.

marcoandredinis replied:
Yeah, I saw that too when upgrading the teleport in teleport-plugins to v13

marcoandredinis commented on integrations/lib/testing/integration/integration_test.go:
```diff
@@ -40,7 +40,7 @@ func (s *IntegrationSuite) TestVersion() {
 	versionMin, err := version.NewVersion("v11.0.0")
 	require.NoError(t, err)
-	versionMax, err := version.NewVersion("v13")
+	versionMax, err := version.NewVersion("v14")
 	require.NoError(t, err)
 
 	assert.True(t, s.Integration.Version().GreaterThanOrEqual(versionMin))
```
I think we could even remove this because we will be always using the branch's version

justinas replied:
In theory this is still useful when running locally, as it will first default to whatever `teleport` you may have in your `PATH`.

marcoandredinis commented on .github/workflows/unit-tests-integrations.yaml:
```diff
@@ -44,7 +44,6 @@ jobs:
       options: --cap-add=SYS_ADMIN --privileged
       env: 
-        TELEPORT_GET_VERSION: v12.1.0
         TELEPORT_ENTERPRISE_LICENSE: ${{ secrets.TELEPORT_ENTERPRISE_LICENSE }}
```
Can we remove this as well? Or is it needed for something else?

marcoandredinis reviewed (approved):
Left some minor comments

Commit:
Remove license from OSS plugins tests

We now run tests against OSS Teleport in the OSS repo,
and against Enterprise Teleport in the E repo.
So the assumption that CI always needs an enterprise license
is incorrect.

zmb3 commented on .github/workflows/unit-tests-integrations.yaml:
```diff
@@ -54,6 +51,9 @@ jobs:
         uses: ./.github/actions/prepare-workspace
 
+      - name: Build teleport binaries
+        run: make && echo "$PWD/build" >> "$GITHUB_PATH"
```
If you don't need `tsh` or `tbot`, you could save some time with:
```suggestion
        run: make build/teleport build/tsh && echo "$PWD/build" >> "$GITHUB_PATH"
```

zmb3 commented on integrations/lib/testing/integration/integration_test.go:
```diff
@@ -40,7 +40,7 @@ func (s *IntegrationSuite) TestVersion() {
 	versionMin, err := version.NewVersion("v11.0.0")
 	require.NoError(t, err)
-	versionMax, err := version.NewVersion("v13")
+	versionMax, err := version.NewVersion("v14")
```
Do we have to update this with each major version?

marcoandredinis replied:
Yes, otherwise tests will fail
Maybe we can get the correct version from `/version.go` and then increment by one
Increment by one because that's actually our compatibility guidelines, right? Clients must support Auth/Proxy with vX+1 (where X is the client's version)

Commit:
Only compile the required binaries
